int32 x
int32 y
int32 z
---
bool isLegal